project/__init__.py

```diff
@@ -1,7 +1,24 @@
 from __future__ import absolute_import, unicode_literals
 
+import warnings
+from pathlib import Path
+
 # This will make sure the app is always imported when
 # Django starts so that shared_task will use this app.
 from .celery import app as celery_app
 
 __all__ = ["celery_app"]
+
+
+try:
+    from dotenv import load_dotenv
+except ImportError:
+    warnings.warn(
+        "Module dotenv not available; not importing local settings from .env",
+        ImportWarning,
+    )
+else:
+    # Load .env variables here, *before* setting remaining values, so that e.g. DEBUG from .env is used to
+    # determine other values. Note this will not overwrite any system-level environment variables (e.g. in .bashrc)
+    env_path = Path(__file__).resolve().parents[1] / ".env"
+    load_dotenv(dotenv_path=env_path, verbose=True)
```

Comment:
does `find_dotenv` not work?

```python
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())
```

Otherwise, why not
```python
Path("..") / ".env"
```
?

Reply:
Using find_dotenv now - thanks! but the reason not to do `Path("..") / ".env"` is that `.` refers to the current working directory, *not* the file, so the path would become dependent on where this file was run. Demo where `project/testfile.py` just prints out `Path(".").resolve()`: - 

```
(lookit-api) bash-3.2$ python project/testfile.py
/Users/kms/lookit-api
(lookit-api) bash-3.2$ cd ~
(lookit-api) bash-3.2$ python lookit-api/project/testfile.py
/Users/kms
(lookit-api) bash-3.2$ 
```

Reply:
Ah, right - in that case, I'd probably pass [`sys.argv[0]`](https://docs.python.org/3/library/sys.html#sys.argv) since it's guaranteed to be the location of the manager script (`__file__` can be weird sometimes, particularly on windows though that doesn't affect us much)

Reply:
Hmm that "guarantee" seems like it depends on the details of how this gets called which might lead to hard-to-reason-about deployment weirdness (e.g., this runs fine locally but not on Travis build). Proposing to stick with __file__ for now given this isn't broken / isn't a change from the Django boilerplate.